Week2/MAKEME.md

```diff
@@ -18,36 +18,110 @@ implement them:
 
 *Read how to [use and escape query values](https://github.com/mysqljs/mysql#escaping-query-values)*
 
-```js
-class TodoModel {
-
-    // ...
+// This is the connector (also known as driver)
+// that we can use to connect to a MySQL process
+// and access its databases.
+const mysql = require('mysql');
 
-    create(description, callback) {
-        // Write code and query to create a new TODO item
+class TodoModel {
+    constructor(dbConnection) {
+        this.dbConnection = dbConnection;
     }
 
-    update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+    // Loads all the TODOs in the database
+    load(callback) {
+        const selectTodoItems = "SELECT * FROM todo_items";
+        this.dbConnection.query(selectTodoItems, function (err, results, fields) {
+            if (err) {
+                callback(err);
+                return;
+            }
+
+            callback(null, results);
+        });
     }
 
-    delete(id, callback) {
-        // Write code and query to delete an existing TODO item
-    }
 
-    tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
-    }
-        
-    untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
-    }
 
-    markCompleted(todoItemId, callback) {
-        // Write code to mark a TODO item as completed
-    }
+
+    function create(id, description, callback) {
+    // Write code and query to create a new TODO item
+    var query = dbConnection.query('INSERT INTO todo_items (text, user_id) VALUES ( ' + description + ' , ' + id + ' )');
```

Comment:
Good job here - but I think the sample code in the other file would have helped a lot.

In general, composing a string (in this case, the string that will become the query) in this way isn't the best. You've used 'string concatenation', where you add two strings together with '+'. It makes it a bit difficult to read, and it's also open to SQL injection vulnerabilities.

You could also use 'string templating', where the string is written with 'backticks', the ` character. When you do that, you can write a string like this:
```
db.Connection.query(`INSERT INTO todo_items (text, user_id) VALUES ( ${ description }, ${ id } )
```
String templating like this is still open to a SQL injection vulnerability. To fix that, you can use `connection.escape()` on the strings to escape them, or use a 'prepared statement'.